# Analysis and Prediction GMSL

## Sea Level Rise Dataset (https://datahub.io/core/sea-level-rise)

Ovaj skup podataka sadrži promenu nivoa mora od 1880. do 2013. godine primećenu satelitima.

Podaci koji su od interesa za naš rad su:
* Date - datum merenja 
* GMSL - globalna srednja vrednost nivoa mora

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('original_datasets/csiro_recons_gmsl_mo.csv', delimiter=',')

In [3]:
df.head()

Time   GMSL  GMSL uncertainty
0  1880-01-15 -183.0              24.2
1  1880-02-15 -171.1              24.2
2  1880-03-15 -164.3              24.2
3  1880-04-15 -158.2              24.2
4  1880-05-15 -158.7              24.2

In [4]:
df.columns

Index(['Time', 'GMSL', 'GMSL uncertainty'], dtype='object')

In [5]:
df.drop('GMSL uncertainty', axis = 1, inplace = True)

In [6]:
df.rename(columns={'Time':'Date'}, inplace = True)

In [7]:
df.head()

Date   GMSL
0  1880-01-15 -183.0
1  1880-02-15 -171.1
2  1880-03-15 -164.3
3  1880-04-15 -158.2
4  1880-05-15 -158.7

In [8]:
df.describe()

GMSL
count  1608.000000
mean    -66.079540
std      62.893171
min    -184.500000
25%    -117.100000
50%     -74.000000
75%     -15.375000
max      82.400000

In [9]:
df.isna().sum()

Date    0
GMSL    0
dtype: int64

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1608 entries, 0 to 1607
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Date    1608 non-null   object 
 1   GMSL    1608 non-null   float64
dtypes: float64(1), object(1)
memory usage: 25.2+ KB


In [11]:
df['Date'] = pd.to_datetime(df['Date'])

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1608 entries, 0 to 1607
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype         
---  ------  --------------  -----         
 0   Date    1608 non-null   datetime64[ns]
 1   GMSL    1608 non-null   float64       
dtypes: datetime64[ns](1), float64(1)
memory usage: 25.2 KB


In [13]:
df['Date'].dt.year

0       1880
1       1880
2       1880
3       1880
4       1880
        ... 
1603    2013
1604    2013
1605    2013
1606    2013
1607    2013
Name: Date, Length: 1608, dtype: int64

In [14]:
#df = df.set_index('Date')

#### Pomoću metode nunique proverićemo da li imamo duplikate za kolonu datum.

In [15]:
df.shape

(1608, 2)

In [16]:
df.nunique()

Date    1608
GMSL    1129
dtype: int64

#### Pošto skup podataka ima 1608 redova, a rezultat nunique vraća da postoji 1608 različitih Date zaključujemo da ne postoje duplikati.

In [17]:
# df['Date'].value_counts() # Iskoristiti da se prebroje vrednosti GMSL kada budu True/False tj. 0/1

In [18]:
#merged_df = df.merge(df_for_merge,left_on = 'State', right_on = 'State')
#merged_df.head()

In [19]:
df.to_csv('processed_datasets/GMSL.csv', index = False)